circuit_knitting/cutting/cutting_experiments.py

```diff
@@ -389,3 +371,80 @@ def _get_pauli_indices(cog: CommutingObservableGroup) -> list[int]:
     if not pauli_indices:
         pauli_indices = [0]
     return pauli_indices
+
+
+def _consolidate_resets(circuit: QuantumCircuit, inplace=True) -> QuantumCircuit:
+    """Consolidate redundant resets into a single reset."""
+    if not inplace:  # pragma: no cover
+        circuit = deepcopy(circuit)
+
+    # Keep up with whether the previous instruction on a given qubit was a reset
+    resets = {i: False for i in range(circuit.num_qubits)}
+
+    # Remove resets which are immediately following other resets
+    remove_ids = []
+    for i, inst in enumerate(circuit.data):
+        qargs = [circuit.find_bit(q).index for q in inst.qubits]
+        if inst.operation.name == "reset":
+            if resets[qargs[0]]:
+                remove_ids.append(i)
+            else:
+                resets[qargs[0]] = True
+        else:
+            for q in qargs:
+                resets[q] = False
```

Comment:
I may be missing something here but does the code block in lines 394-395 need to be there given that in line 382, `resets` is initialized to all its elements being `False`?

Reply:
Yes, because a qubit could have been flipped to `True` at some point. This line is saying that we have a non-reset instruction and to flip all of the associated qargs back to False (indicating that the most recent instruction on that qubit was not a reset)

Reply:
I see, that makes sense, thanks.